src/librustc_resolve/lib.rs

```diff
@@ -1183,7 +1182,7 @@ impl<'a> Resolver<'a> {
                 vis: ty::Visibility::Public,
             }),
 
-            crate_loader,
+            crate_loader: CrateLoader::new(session, cstore, crate_name),
```

Comment:
I don't like this.
`CrateLoader` is an "immutable reference" to some fields in `rustc_interface::Compiler` plus the result of `rustc_interface` query `crate_name()`.
(`metadata_loader` should also migrate from `CStore` to `CrateLoader` and it's also a kind of "immutable reference" to `rustc_interface::Compiler`.)
`CrateLoader::new` is called from multiple places right now, but it should always return the same thing.
I'll probably try to add a `rustc_interface` query for it.

I want to split the crate locator configuration (`locator::Context`) into three parts - immutable per-session configuration, immutable per-search configuration, and mutable output / in-progress state.
`CrateLoader` is supposed to be the "immutable per-session configuration" in this case, that's why I want to make sure that it's indeed the same all the time.

Reply:
I've reverted this bit, `CrateLoader` is now constructed in the same place as before.
I'll try to do something with it later.